rasa/graph_components/validators/default_recipe_validator.py

```diff
@@ -0,0 +1,505 @@
+            )
+
+    def _warn_of_competition_with_regex_extractor(
+        self, training_data: TrainingData
+    ) -> None:
+        """Warns when regex entity extractor is competing with a general one.
+
+        This might be the case when the following conditions are all met:
+        * You are using a general entity extractor and the `RegexEntityExtractor`
+        * AND you have regex patterns for entity type A
+        * AND you have annotated text examples for entity type A
+
+        Args:
+            training_data: The training data for the NLU components.
+        """
+        present_general_extractors = self._component_types.intersection(
+            TRAINABLE_EXTRACTORS
+        )
+        has_general_extractors = len(present_general_extractors) > 0
+        has_regex_extractor = (
+            RegexEntityExtractorGraphComponent in self._component_types
+        )
+
+        regex_entity_types = {rf["name"] for rf in training_data.regex_features}
+        overlap_between_types = training_data.entities.intersection(regex_entity_types)
+        has_overlap = len(overlap_between_types) > 0
+
+        if has_general_extractors and has_regex_extractor and has_overlap:
+            rasa.shared.utils.io.raise_warning(
+                f"You have an overlap between the "
+                f"{RegexEntityExtractorGraphComponent.__name__} and the "
+                f"statistical entity extractors "
+                f"{_types_to_str(present_general_extractors)} "
+                f"in your configuration. Specifically both types of extractors will "
+                f"attempt to extract entities of the types "
+                f"{', '.join(overlap_between_types)}. "
+                f"This can lead to multiple "
+                f"extraction of entities. Please read "
+                f"{RegexEntityExtractorGraphComponent.__name__}'s "
+                f"documentation section to make sure you understand the "
+                f"implications.",
+                docs=f"{DOCS_URL_COMPONENTS}#regexentityextractor",
+            )
+
+    def _raise_if_featurizers_are_not_compatible(self,) -> None:
+        """Raises or warns if there are problems regarding the featurizers.
+
+        Raises:
+            `InvalidConfigException` in case the featurizers are not compatible
+        """
+        featurizers: List[SchemaNode] = [
+            node
+            for node in self._graph_schema.nodes.values()
+            if issubclass(node.uses, Featurizer2)
+        ]
+        Featurizer2.raise_if_featurizer_configs_are_not_compatible(
+            [schema_node.config for schema_node in featurizers]
+        )
+
+    def _validate_core(self, story_graph: StoryGraph, domain: Domain) -> None:
+        """Validates whether the configuration matches the training data.
+
+        Args:
+           story_graph: a story graph (core training data)
+           domain: the domain
+        """
+        if not self._policy_schema_nodes and story_graph.story_steps:
```

Comment:
I think that if it is necessary that we do that validation in the default recipe validation, then we can keep it there -- but it looks like it isn't necessary and so I'd rather keep it here because I'd expect the (complete) validation logic inside the validation component and would be surprised later if part of the cases is handled elsewhere. 
Also this validator is unaware of the training type which (I'm guessing here) is no replacement for the "what training data do we have" but for the CLI mode? Wdyt? Shall I merge as is for now? (or add training type flag to the validation here?)